src/org/labkey/test/components/ui/navigation/apps/ServerNotificationMenu.java

```diff
@@ -0,0 +1,201 @@
+
+        if(status.contains("fa-bell"))
+        {
+            return LoadingStatus.COMPLETE;
+        }
+        else
+        {
+            return LoadingStatus.RUNNING;
+        }
+    }
+
+    /**
+     * Is the 'Mark all as read' link visible?
+     *
+     * @return True if visible, false otherwise.
+     */
+    public boolean isMarkAllVisible()
+    {
+        return elementCache().markAll().isDisplayed();
+    }
+
+    /**
+     * Click the 'Mark all as read' link. This will cause the list to refresh, so return a new reference to the
+     * notification list.
+     *
+     * @return A new list of {@link ServerNotificationItem}.
+     */
+    public List<ServerNotificationItem> clickMarkAll()
+    {
+        elementCache().markAll().click();
+        return getNotifications();
+    }
+
+    /**
+     * Expand the menu and return the notification.
+     *
+     * @return A list of {@link ServerNotificationItem}, empty if there are none.
+     */
+    public List<ServerNotificationItem> getNotifications()
+    {
+        if(!isExpanded())
+            expand();
+
+        if(getWrapper().isElementPresent(elementCache().notificationsContainerLocator))
+        {
+            return elementCache().notifications();
+        }
+        else
+        {
+            return new ArrayList<>();
+        }
+
+    }
+
+    /**
+     * If no notifications get the message that is present.
+     *
+     * @return Message shown if there are no notifications, and empty string of there are notifications.
+     */
+    public String getNoNotificationsMessage()
+    {
+
+        if(!isExpanded())
+            expand();
+
+        if(elementCache().noNotificationsElement().isDisplayed())
+        {
+            return elementCache().noNotificationsElement().getText();
+        }
+        else
+        {
+            return "";
+        }
+
+    }
+
+    @Override
+    protected Locator getToggleLocator()
+    {
+        return Locator.tagWithId("button", "server-notifications-button");
+    }
+
+    @Override
+    protected ElementCache elementCache()
+    {
+        return (ElementCache) super.elementCache();
+    }
+
+    @Override
+    protected ElementCache newElementCache()
+    {
+        return new ElementCache();
+    }
+
+    protected class ElementCache extends BaseBootstrapMenu.ElementCache
+    {
+        private final Locator notificationsContainerLocator = Locator.tagWithClass("div", "server-notifications-listing-container");
+
+        public final WebElement notificationsContainer()
+        {
+            return notificationsContainerLocator.findWhenNeeded(this);
+        }
+
+        public final List<ServerNotificationItem> notifications()
+        {
+            return new ServerNotificationItem.AppNotificationEntryFinder(getDriver()).findAll(notificationsContainer());
+        }
+
+        public final WebElement statusIcon()
+        {
+            return Locator.tagWithClass("i", "fa").findWhenNeeded(this);
+        }
```

Comment:
This locator could be a bit more specific.
```suggestion
        public final WebElement statusIcon()
        {
            return Locator.byClass("navbar-header-icon").findWhenNeeded(this);
        }
```

Reply:
Changed.